### DISCUSSION TITLE
ドキドキ☆雑談スレッド

### DISCUSSION THREAD

**Discussion by hayashi-ay** [katataku/minishell#64] [General]
part2

part1: https://github.com/katataku/minishell/discussions/6

**Comment by hayashi-ay** [comment]
shellcheckを入れてもらうと幸せになれるかもです。VSCodeのプラグインにあると思います。

**Reply by katataku** [reply]
とりあえず入れました！今日話す時かるーく使い方デモしてもらえるとうれしいです！

**Reply by hayashi-ay** [reply]
okです〜

**Comment by hayashi-ay** [comment]
envでの引数の対応ができてなさそう。

```
env PWD
```

**Reply by hayashi-ay** [reply]
課題要件的には対応しないと行けなさそう。
```
env with no options or arguments
```

**Reply by hayashi-ay** [reply]
解釈的には`env without options or arguments`みたい。なので対応しなくて良さげ。

**Reply by katataku** [reply]
対応不要！

**Comment by hayashi-ay** [comment]
```bash
bash-3.2$ env | grep PWD
PWD=/Users/ahayashi
bash-3.2$ env PWD
/Users/ahayashi
bash-3.2$ echo $PWD
/Users/ahayashi
bash-3.2$ unset PWD
bash-3.2$ env | grep PWD
bash-3.2$ env PWD
/Users/ahayashi
bash-3.2$ echo $PWD

```

envのこの挙動が良くわからない。unsetしてるはずなのにenv PWDだと出力できちゃう。

**Reply by katataku** [reply]
https://atmarkit.itmedia.co.jp/ait/articles/1801/18/news016.html
envコマンド実行時に「環境変数に特定の値を指定しつつ、コマンドを実行」の動きになってる？

**Reply by hayashi-ay** [reply]
参照するのはbashのリファレンスが良いと思います。

**Reply by katataku** [reply]
builtinのenvじゃなくて/usr/bin/envが呼ばれてる可能性もあると思ってますー

**Reply by hayashi-ay** [reply]
な、なるほど〜〜〜〜。たしかに。自前で管理してるけど大丈夫なのかな？？

**Comment by hayashi-ay** [comment]
@katataku もしかして通知がいってないのでメンション

**Comment by hayashi-ay** [comment]
たまに初回のコマンド実行でexecveのエラーでるときあるんですよね〜。

<img width="279" alt="スクリーンショット 2022-03-02 18 08 35" src="https://user-images.githubusercontent.com/39038230/156330520-2ada7daf-d829-4425-92e3-db474a1aa5fc.png">

**Reply by katataku** [reply]
なんでだろ。
手元で再現しなかったので、再現待ちにしたいですね。

頻繁に起きますか？

**Reply by hayashi-ay** [reply]
たまに起きるんですよね〜。再現確認しようとして同じコマンドを試しても再現しないんですけど。

**Comment by katataku** [comment]
バグ見つけましたーー

[事象]以下のコマンド入力時の挙動が違う
`ls -"la"`

[想定動作]
```bash
$ ls -"la"
total 1192
drwxr-x---+  82 taku  staff    2624  3  3 09:51 .
drwxr-xr-x    5 root  admin     160  2  7 06:22 ..
```

[実動作]
```bash
> ls -"la"
ls: -: No such file or directory
ls: la: No such file or directory
```

**Reply by katataku** [reply]
空白文字もトークンとして認識してからparseしないといけないかも？

**Reply by hayashi-ay** [reply]
```
$ ./minishell 
> ls a"aa"
ls: a: No such file or directory
ls: aa: No such file or directory
```

単語の間にダブルクオートなどがあると2つとして認識してしまう感じなんですね。

**Reply by hayashi-ay** [reply]
バグ用discussion or Issueに立てても良いかも。

**Reply by katataku** [reply]
プロジェクトボードにタスクとして積んで、他のタスクと同じ運用にしましょ〜

**Reply by hayashi-ay** [reply]
でもタスクとして積む前にバグかどうかの議論したくないですか？

**Reply by katataku** [reply]
それは確かに！
issueに立てるとタスクとして決まった感じするので、バグ用discussionにしましょ〜

**Reply by hayashi-ay** [reply]
🦋 🐛 🐝 🐞 🐜 🕷 🕸 🦂 🦗 🦟

**Comment by hayashi-ay** [comment]
GithubのPRの差分ファイル表示がtree viewに対応したので結構PR見るの楽になったかも

<img width="1404" alt="スクリーンショット 2022-03-04 11 50 41" src="https://user-images.githubusercontent.com/39038230/156707162-a871fa0a-cfd9-4d10-bba7-fdf2d106ff74.png">

**Reply by katataku** [reply]
いいですね！enableにしました！

**Comment by hayashi-ay** [comment]
ブランチ名にIssue番号を入れているのでPR時に自動でIssueと紐付けるActionを作ってもいいかも。探せばあるかも。

**Comment by hayashi-ay** [comment]
utilsモジュールを作っちゃいます。

**Comment by hayashi-ay** [comment]
使ってなさそう。

https://github.com/katataku/minishell/blob/main/includes/define.h#L22

**Comment by hayashi-ay** [comment]
そういえば今lexerの段階でWORDがクオートに囲われているかどうかという情報が失われちゃってますね。

**Reply by hayashi-ay** [reply]
いやトークンにあるか

**Comment by katataku** [comment]
こさかさんと話した内容に対応するか。

```bash
$?の戻り値が特殊
zz | echo $?
echo $? | zz
```

**Reply by hayashi-ay** [reply]
しましぇ〜〜〜〜ん

**Reply by hayashi-ay** [reply]
zshはまた挙動が違うのとドキュメント的には以下なので。

> ($?) Expands to the exit status of the most recently executed foreground pipeline.

**Comment by hayashi-ay** [comment]
バッククオートって扱いどうしましょう？いまlexer, parserでは考慮してますけど、expansionでは考慮しない予定です。なのでlexerで通常文字として扱ってもいいかなと思います。
